test/units/modules/test_apt.py

```diff
@@ -1,45 +1,57 @@
+# Copyright: Contributors to the Ansible project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 from __future__ import annotations
 
 import collections
 
-from unittest.mock import Mock
-import unittest
-
 from ansible.modules.apt import (
     expand_pkgspec_from_fnmatches,
+    DPKG_OPTIONS,
+    expand_dpkg_options,
 )
+import pytest
+
+FakePackage = collections.namedtuple("Package", ("name",))
+fake_cache = [
+    FakePackage("apt"),
+    FakePackage("apt-utils"),
+    FakePackage("not-selected"),
+]
+
+
+@pytest.mark.parametrize(
+    ("test_input", "expected"),
+    [
+        pytest.param(
+            ["apt"],
+            ["apt"],
+            id="trivial",
+        ),
+        pytest.param(
+            ["apt=1.0*"],
+            ["apt=1.0*"],
+            id="version_wildcard",
+        ),
+        pytest.param(
+            ["apt*=1.0*"],
+            ["apt", "apt-utils"],
+            id="pkgname_wildcard_version_wildcard",
+        ),
+        pytest.param(
+            ["apt*"],
+            ["apt", "apt-utils"],
+            id="pkgname_expands",
+        ),
+    ],
+)
+def test_expand_pkgspec_from_fnmatches(test_input, expected):
+    """Test expand_pkgspec_from_fnmatches."""
```

Comment:
Let's clarify at list a little.
```suggestion
    """Test positive cases of ``expand_pkgspec_from_fnmatches``."""
```